runtime/compiler/control/HookedByTheJit.cpp

```diff
@@ -425,6 +428,42 @@ static void jitHookInitializeSendTarget(J9HookInterface * * hook, UDATA eventNum
          }
       }
 
+#if defined(J9VM_OPT_JITSERVER)
+   bool methodCachedAtServer = false;
+   PersistentUnorderedSet<std::string> *serverAOTMethodSet =
+      (PersistentUnorderedSet<std::string> *) jitConfig->serverAOTMethodSet;
+
+   if (serverAOTMethodSet != NULL)
+      {
+      // Construct a signature
+      J9UTF8 *className;
+      J9UTF8 *name;
+      J9UTF8 *signature;
+      getClassNameSignatureFromMethod(method, className, name, signature);
+      // SigLen calculation as used in TRJ9VMBase::printTruncatedSignature
+      int32_t sigLen = J9UTF8_LENGTH(className) + J9UTF8_LENGTH(name) +
+         J9UTF8_LENGTH(signature) + 2;
+
+      char *sigC = new char[sizeof(char) * (sigLen + 1)];
```

Comment:
OpenJ9 has its own memory management that provides accounting, so using the system allocator is not a good choice. On the other hand we want to minimize the usage of persistent allocator because of the overhead and fragmentation it can create. Here's an example on how this achieved in other parts of the code: https://github.com/eclipse-openj9/openj9/blob/master/runtime/compiler/control/HookedByTheJit.cpp#L180
Basically we use the stack for most methods, and for the few very long signatures we use the persistent allocator. In this case I think that we can not set a low count for methods that have a very long name. These should be very few.